resource/resourceadapters/charmstore.go

```diff
@@ -110,7 +119,8 @@ func (client CSRetryClient) GetResource(req charmstore.ResourceRequest) (charmst
 	var lastErr error
 	args.NotifyFunc = func(err error, i int) {
 		// Remember the error we're hiding and then retry!
-		logger.Debugf("(attempt %d) retrying resource download from charm store due to error: %v", i, err)
+		logger.Warningf("(attempt %d) retrying resource %q download from charm store [channel (%v), charm (%v), resource revision (%v)] due to error: %v",
```

Comment:
"attempt %d to download resource %q from charm store [channel (%v), charm (%v), resource revision (%v)] failed with error (will retry): %v"

I'm not sure if its all better, but it reads slightly better for me.